v2/encoding.go

```diff
@@ -30,12 +30,15 @@ func (e Encoding) String() string {
 type xencodingWrapper struct {
 	decoder *xencoding.Decoder
 	encoder *xencoding.Encoder
+
+	encoding xencoding.Encoding
 }
```

Comment:
I suggest to complete delete `xencodingWrapper`, use just `xencoding.Encoding` and create new encoder/decoder every time we need to encode/decode.

xencodingWrapper is used to just micro-optimise performance but apparently it just introduces issues